reco_utils/dataset/pandas_df_utils.py

```diff
@@ -166,3 +166,93 @@ def _convert(field, feature, field_index, field_feature_index_dict):
     return df_new
 
 
+def negative_feedback_sampler(
```

Comment:
Maybe it also makes sense to have a threshold option. I.e. if threshold == 2 any ratings below 2 will be considered negative; if you have 100 ratings >= 2, 70 ratings < 2 then you can fill in the remaining 30 negatives with sampling.

Reply:
Yea plan to add these (maybe in another PR). We have these in PySpark implementations.

Reply:
Good point. Do we want to have another function for this? Another thing is that the negative sampler in this PR does not assume an existence of rating column, meaning that, the negative samples (to a user) are defined as those items that have not been interacted with the user before.

Reply:
Right, better leave this for another function. I guess there is some common code to be reused.